js/researches/english/EnglishParticiple.js

```diff
@@ -0,0 +1,158 @@
+ */
+var isPrecededByDeterminer = function( participle, word, sentencePart ) {
+	if ( participle !== word ) {
+		return false;
+	}
+	var indexOfFit = sentencePart.indexOf( word );
+	var determinerMatches = getWordIndices( sentencePart, determinersRegex );
+	return includesIndex( determinerMatches, indexOfFit );
+};
+
+/**
+ * Creates an Participle object for the English language.
+ *
+ * @param {string} participle The participle.
+ * @param {string} sentencePart The sentence part that contains the participle.
+ * @param {object} attributes  The attributes object.
+ *
+ * @constructor
+ */
+var EnglishParticiple = function( participle, sentencePart, attributes ) {
+	Participle.call( this, participle, sentencePart, attributes );
+	this.checkException();
+};
+
+require( "util" ).inherits( EnglishParticiple, Participle );
+
+/**
+ * Sets sentence part passiveness to passive if there is no exception.
+ * @returns {void}
+ */
+EnglishParticiple.prototype.checkException = function() {
+	if ( isEmpty( this.getParticiple() ) ) {
+		this.setSentencePartPassiveness( false );
+		return;
+	}
+	var isPassive =
+		! this.isNonVerbEndingEd() &&
+		! this.hasRidException() &&
+		! this.hasHavingException() &&
+		! this.hasLeftException() &&
+		! this.hasFitException();
+	this.setSentencePartPassiveness( isPassive );
+};
+
+/**
+ * Checks whether a found participle is in the nonVerbsEndingEd list.
+ * If a word is in the nonVerbsEndingEd list, it isn't a participle.
+ * Irregular participles do not end in -ed, and therefore cannot be in the nonVerbsEndingEd list.
+ *
+ * @returns {boolean} Returns true if it is in the nonVerbsEndingEd list, otherwise returns false.
+ */
+EnglishParticiple.prototype.isNonVerbEndingEd = function() {
+	if ( this.getType() === "irregular" ) {
+		return false;
+	}
+	var participle = this.getParticiple();
+	return includes( nonVerbsEndingEd, participle );
+};
+
+/**
+ * Checks whether the participle is 'rid' in combination with 'get', 'gets', 'getting', 'got' or 'gotten'.
+ * If 'rid' appears in combination with 'get', 'gets', 'getting', 'got' or 'gotten', it is not passive.
+ *
+ * @returns {boolean} Returns true if 'rid' is found in combination with 'get', 'gets', 'getting', 'got' or 'gotten',
+ * otherwise returns false.
+ */
+EnglishParticiple.prototype.hasRidException = function() {
+	var participle = this.getParticiple();
+	if ( participle === "rid" ) {
+		var auxiliaries = this.getAuxiliaries();
+		return  ! isEmpty( intersection( irregularExclusionArray, auxiliaries ) );
+	}
+	return false;
+};
+
+/**
+ * Checks whether the participle is directly preceded by 'having'.
+ * If this is the case, the sentence part is not passive.
+ *
+ * @returns {boolean} Returns true if having is directly preceding the participle, otherwise returns false.
+ */
+EnglishParticiple.prototype.hasHavingException = function() {
+	var participle = this.getParticiple();
+	var sentencePart = this.getSentencePart();
```

Comment:
SentencePart is used in more than 1 place, so defining it as a variable seems good in this place.